pubsub/tests/system.py

```diff
@@ -187,12 +187,14 @@ class AckCallback(object):
     def __init__(self):
         self.calls = 0
         self.lock = threading.Lock()
+        self.data = {}
 
     def __call__(self, message):
         message.ack()
         # Only increment the number of calls **after** finishing.
         with self.lock:
             self.calls += 1
+            self.data[message.attributes['num']] = message.data
```

Comment:
But why?

Reply:
Could it be possible that I didn't get all of the messages at least once but have multiple duplicates?  I wasn't sure.